### PR TITLE
CBG-1387 / CBG-1411: Channel Access Revocation for ISGR

### PR DESCRIPTION
**Pull request by JRascagneres**
- Removed enable_auto_purge config option and use the existing purge on removal option
- Utilize changes response

Tested with small unit test

Currently doesn't cover no rev case. Will follow up in a different PR with this - CBG-1477

### CODE REVIEW

**Review by bbrks — CHANGES_REQUESTED** [review]

File: db/blip_handler.go [hunk 1]
```diff
@@ -477,7 +477,20 @@ func (bh *blipHandler) handleChanges(rq *blip.Message) error {
...
+		deletedFlags := changesDeletedFlag(0)
+		// TODO: Add check for Blip Protocol Version when CBG-1435 is in
+		if len(change) > 3 {
+			var err error
+			deletedIntFlag, err := change[3].(json.Number).Int64()
+			if err != nil {
+				base.InfofCtx(bh.loggingCtx, base.KeyAll, "Failed to parse deletedFlags: %v", err)
+				continue
+			}
+			deletedFlags = changesDeletedFlag(deletedIntFlag)
```
[the remark this answers is not in the document]
Comment (JRascagneres) [reply db/blip_handler.go]: Yeah, thats a good plan, will implement this. I don't think we want to return an error though in the error case. I think we want to 'continue' so we continue iterating over the other changes messages.

File: rest/replicator_test.go [hunk 2]
```diff
@@ -4883,6 +4883,64 @@ func TestReplicatorConflictAttachment(t *testing.T) {
...
+	})
+	defer rt1.Close()
+
+	revocationTester.addRole("user", "foo")
+	revocationTester.addRoleChannel("foo", "chanA")
+
+	resp := rt2.SendAdminRequest("PUT", "/db/doc1", `{"channels": "chanA"}`)
+	assertStatus(t, resp, http.StatusCreated)
+
+	srv := httptest.NewTLSServer(rt2.TestPublicHandler())
```
Comment (bbrks) [inline rest/replicator_test.go]:
```suggestion
	srv := httptest.NewServer(rt2.TestPublicHandler())
```

File: rest/replicator_test.go [hunk 3]
```diff
@@ -4883,6 +4883,64 @@ func TestReplicatorConflictAttachment(t *testing.T) {
...
+
+	ar := db.NewActiveReplicator(&db.ActiveReplicatorConfig{
+		ID:          t.Name(),
+		Direction:   db.ActiveReplicatorTypePull,
+		RemoteDBURL: passiveDBURL,
+		ActiveDB: &db.Database{
+			DatabaseContext: rt1.GetDatabase(),
+		},
+		Continuous:          false,
+		InsecureSkipVerify:  true,
```
Comment (bbrks) [inline rest/replicator_test.go]:
```suggestion
```

File: db/blip_handler.go [hunk 4]
```diff
@@ -477,7 +477,27 @@ func (bh *blipHandler) handleChanges(rq *blip.Message) error {
 		if nWritten > 0 {
 			output.Write([]byte(","))
 		}
-		if missing == nil {
+
+		deletedFlags := changesDeletedFlag(0)
+		if len(change) > 3 {
+			switch v := change[3].(type) {
+			case json.Number:
+				deletedIntFlag, err := change[3].(json.Number).Int64()
```
Comment (bbrks) [inline db/blip_handler.go]:
```suggestion
				deletedIntFlag, err := v.Int64()
```

**Review by bbrks — CHANGES_REQUESTED** [review]

File: db/blip_handler.go [hunk 5]
```diff
@@ -477,7 +477,27 @@ func (bh *blipHandler) handleChanges(rq *blip.Message) error {
...
+			case bool:
+				deletedFlags = changesDeletedFlagDeleted
+			default:
+				base.ErrorfCtx(bh.loggingCtx, "Unknown type for deleted field in changes message: %T", v)
+				continue
+			}
+
+		}
+
+		if missing == nil && deletedFlags&changesDeletedFlagRevoked == 0 && deletedFlags&deletedFlags&changesDeletedFlagRemoved == 0 {
```
Comment (bbrks) [inline db/blip_handler.go]:
```suggestion
		if missing == nil && deletedFlags&changesDeletedFlagRevoked|changesDeletedFlagRemoved == 0 {
```

**Review by bbrks — CHANGES_REQUESTED** [review]
maybe this too?

**Review by bbrks — DISMISSED** [review]

**Review by bbrks — APPROVED** [review]

**Review by adamcfraser — APPROVED** [review]

**Pull request merged**